src/components/WalletDropdown/MiniPortfolio/Activity/parseLocal.ts

```diff
@@ -182,24 +181,22 @@ export function parseLocalActivity(
   return { ...defaultFields, ...additionalFields }
 }
 
-export function useLocalActivities(): ActivityMap | undefined {
+export function useLocalActivities(account: string): ActivityMap {
   const allTransactions = useMultichainTransactions()
-  const { chainId } = useWeb3React()
   const tokens = useCombinedActiveList()
 
   return useMemo(
     () =>
-      chainId
-        ? allTransactions.reduce((acc: { [hash: string]: Activity }, [transaction, chainId]) => {
-            try {
-              const localActivity = parseLocalActivity(transaction, chainId, tokens)
-              if (localActivity) acc[localActivity.hash] = localActivity
-            } catch (error) {
-              console.error('Failed to parse local activity', transaction)
-            }
-            return acc
-          }, {})
-        : undefined,
-    [allTransactions, chainId, tokens]
+      allTransactions.reduce((acc: { [hash: string]: Activity }, [transaction, chainId]) => {
+        if (transaction.from !== account) return acc
+        try {
+          const localActivity = parseLocalActivity(transaction, chainId, tokens)
+          if (localActivity) acc[localActivity.hash] = localActivity
+        } catch (error) {
+          console.error('Failed to parse local activity', transaction)
```

Comment:
does this need to be an error, or can we use `console.debug`

Reply:
also can you try to cover this line in a test

Reply:
I was able to remove the try/catch it was redundant since everything is strictly typed and we're properly checking undefined from token maps now

Reply:
GH is complaining abt about another line that is missing coverage now but I plan to cover that in [this ticket](https://uniswaplabs.atlassian.net/browse/WEB-3113)